### ISSUE TITLE
Irreproducible results

### ISSUE THREAD

**Issue by susilehtola** [dftlibs/xcfun#163]
XCFun currently relies on compile-time definitions to change functionals' behavior. This means that for instance PBE exchange in two copies of the same version of XCFun may give different results. This is highly undesirable behavior: the output of functionals should not depend on compile-time behavior.

Instead, the variants should be implemented separately, as in Libxc. One should be able to access all implementations without having to reconfigure and recompile in-between.

**Comment by robertodr** [comment]
Are you referring to these lines? https://github.com/dftlibs/xcfun/blob/master/src/functionals/pbex.hpp#L30-L35

**Comment by susilehtola** [comment]
> Are you referring to these lines? https://github.com/dftlibs/xcfun/blob/master/src/functionals/pbex.hpp#L30-L35

Yes. Other functionals have similar issues, e.g. https://github.com/dftlibs/xcfun/blob/6c82faa0b194d27d8f2fcc023b72df905b22ee9d/src/functionals/p86c.cpp#L31-L35

**Comment by ilfreddy** [comment]
@susilehtola it seems we keep stumbling upon this over and over (see https://github.com/MRChemSoft/mrchem/issues/509). Maybe it is about time to fix this compile-time "flavours" of XCFun. Maybe we could link it to our ongoing MRChem/Libxc project.
